external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java

```diff
@@ -107,14 +110,14 @@ public void refresh() {
                         _stormConf,
                         _spoutConfig,
                         id,
-                        deletedManagers.get(id.partition));
+                        deletedManagers.get(new TopicAndPartition(id.topic, id.partition)));
```

Comment:
Are you sure this lookup makes sense? The deletedManagers only contains partitions that this task is no longer responsible for (curr - mine). That partition shouldn't be able to also be in newPartitions (mine - curr). Won't this always return null (unless you're hitting the bug that this PR is trying to fix, namely that offsets are being copied from the PartitionManager for e.g. topic1 to the PartitionManager for topic2)?

Reply:
I believe that this lookup exists for preserving information about partitions from other brokers in case of rebalancing. If for example one of brokers fail and its partitions are reassigned to other brokers this lookup should find information about this partition and reuse it during recreation of the same partition but on other host. ``equals()`` and ``hashCode()`` in Partition are based on topic, partition and host while this lookup in based only on topic and partition.

Reply:
Good point, never mind then.